docs/relnotes/compact/minokawa-0-18-26-0.mdx

```diff
@@ -0,0 +1,418 @@
+  Note that there is generally no corresponding cast in the opposite direction.
+  
+- From `Bytes<n>` to `Vector<n, T>`, a bytes value type to a vector type of the same length, where `T` is a supertype of `Uint<8>`.
+  It was previously possible to cast from `Bytes<n>` to `Vector<n, Uint<8>>`, that is where the vector element type is exactly `Uint<8>`.
+  You can now cast to a vector type whose element type is any supertype of `Uint<8>`.
+  The cast cannot fail at runtime, but it has a cost proportional to the length `n`.
+  This cost is the same as the cost of a cast to `Vector<n, Uint<8>>`.
+  
+- From `Bytes<n>` to `[T1, ..., Tn]`, a bytes value to a tuple type of the same length, where each `Ti` is a (possibly different) supertype of `Uint<8>`.
+  Since vector types are equivalent to a corresponding tuple type, the cast above is a special case of this one.
+  It cannot fail at runtime, but it has a cost proportional to the length `n`.
+
+- From `Uint` types to `Bytes` types.
+  This was previously possible using an intermediate `Field` type.
+  The semantics is the same as casting through an intermediate `Field` type.
+  Specifically, there is a runtime check that the `Bytes` type is big enough to hold the actual `Uint` value, so the cast can fail.
+  There is a representation change, so there is a runtime cost (both in JavaScript and in terms of proof size).
+  
+- From `Bytes` types to `Uint` types.
+  This was previously possible using an intermediate `Field` type.
+  The cast can fail at runtime and it involves a representation change.
+  
+- From `Field` to `enum` types.
+  You could previously cast from `enum` types to `Field` but not vice versa.
+  This cast is checked at runtime, so it can fail.
+  There is a JavaScript representation change (from `bigint` to `number`) but no representation change in the proof (there is still a range check in the proof).
+  
+- From `enum` types to `Uint` types and vice versa
+  These were possible in one direction using an intermediate `Field` and not in the other direction.
+  These casts might involve a runtime check in JavaScript and in the proof.
+  There will be a runtime check for an `enum` to `Uint` cast if the `Uint` type is not big enough to represent all the `enum` values.
+  There will be a runtime check for a `Uint` to `enum` cast if the `enum` type does not have enough values to represent all the values of the `Uint` type.
+  Both directions require a representation change in JavaScript.
+  `Uint` types are represented by the JavaScript type `bignum` and `enum` types are represented by `number`.
+
+These new casts are **non-breaking** changes, we have added new casts but we have not changed the behavior of any of the existing ones.
+
+### We have renamed some functions in the Compact runtime
+
+The compiler-generated JavaScript code uses the Compact runtime to provide implementations of behavior that doesn't depend on the specific contract.
```

Comment:
The interface to these routines has also changed: each takes a third argument, which is a string describing the source location of the cast.